### ISSUE TITLE
[TB] Typography modal values don't match stated values

### ISSUE THREAD

**Issue by aaronreed708** [finos/a11y-theme-builder#310]
## Problem/Concern

When I go to modify the body styles in Typography atoms, for example, and go to edit one of the entries, the defaults on the modal that pops up don't match the current values stated in the entry.

To recreate:

1. Load the pre-defined Sample design system
2. Click on Typography->Body Styles from the left nav bar
3. Click on the Edit button for the "Body 1" style.  Note the currently selected values, like 500 for font-weight

You'll notice that in the modal that pops up that the font-weight selected by default in the dialog is 700.

Illustrations added by @evangk6  in comment below.

## Proposed Solution

The modal values and the stated values should always be in sync.

**Comment by evangk6** [comment]
Selecting `Atoms->Typography->Body Styles` (although the issue will appear with any typography category). Notice the font weight of **500** for `Body 1`. This value is taken from the `sampleRef `.
![image](https://user-images.githubusercontent.com/36457549/234346983-e0fc8c5d-98da-41cd-b149-2fb662661d7f.png)

Opening the edit modal for `Body 1`, we see that the selected weight is **700**. This value is directly from the sdk property.
![image](https://user-images.githubusercontent.com/36457549/234347283-a3dbf8d7-b912-4a78-a0f3-b020ee2629ba.png)

Selecting `Components->Typography->Body` we see that this weight is a third different value of **400**. This value is also from the `sampleRef`.
![image](https://user-images.githubusercontent.com/36457549/234348166-44342d8c-7513-4539-8d5f-ecc863e29eb5.png)

Going back to the Body Atom, if I select another value in the dropdown, in this case, 300. The value now reflects the correct value of **300**.
![image](https://user-images.githubusercontent.com/36457549/234348773-99cce3f1-77d8-4b8e-b9c7-b1dc89c1ae0c.png)

Unfortunately the Body Component still has an unchanged value of **400**.
![image](https://user-images.githubusercontent.com/36457549/234349067-4502e9db-6c52-46c0-aa6d-7ed2fbe5d367.png)

**Comment by evangk6** [comment]
A couple of things I want to note right away that I believe fixes one of the issues:

The `fontWeight` state that is used in the `fontEditModal` is set by default in a different way than the other states
```
const [fontWeight,  setFontWeight ] = useState<number>((fontWeightProperty as any).value || defaultWeight);
```

Changing this back to the below, fixes the issue. I did not make this change so I do not know why it is different for this property.
```
const [fontWeight,  setFontWeight ] = useState<number>(fontWeightProperty.getValue() || defaultWeight);
```

As an aside I also noticed that the `defaultWeight` is being set to the default for every header, when instead it should be set according to the style we are working with. This tells me that the sdk defaults are not set correctly for each `typographyStyling`.

**NOTE:** This does not appear to fix the component issue, however,
